### ISSUE TITLE
Ability to specify relation create new or use existing

### ISSUE THREAD

**Issue by j4k** [laracasts/TestDummy#97]
Hi @JeffreyWay 

If theres a way to do this and I've missed it apologies...

TestDummy creates a new model instance for every relation specified, this is okay, but it would be awesome if there was some way to specify whether existing seeded data should be used to populate the relation based on a previously seeded model.

For example, when using TestDummy for seeding creating 50 of a User factory, 100 of a Project factory, which has two references to User - created_by / completed_by ends up with 250 inserts to Users, for the amount you requested, plus 100 for each of the relations. This grows exponentially when trying to populate pivot tables as well, and doesn't really reflect pivot data that well.

I propose adding another string like `'factory:App\User'` that could be something like `'model:App\User'` that could get a count of existing Models and use Faker's `numberBetween` method to generate a relationship integer, or something to that effect.

I would be happy to submit a PR.

**Comment by ehkasper** [comment]
Yeah, that would be cool. @j4k, do you have any code written yet?

**Comment by j4k** [comment]
Not yet, will submit a PR soon, just wanted to see if people thought it might be useful

**Comment by ven7ura** [comment]
I also see this as a very important feature! I will wait for any update.

**Comment by efrontsa** [comment]
+1

**Comment by efrontsa** [comment]
@j4k @ehkasper @ven7ura 
the work around for it is you can do the following

`$factory(Comment::class, [
    'post_id' => $faker->randomElement(range(1, 20))
]);`

**Comment by phroggyy** [comment]
@j4k, @ehkasper, @ven7ura, @efrontsa I have most of it working on [my fork](https://github.com/phroggyy/TestDummy) and will submit a PR hopefully today, since I don't see it having been implemented yet.

**Comment by phroggyy** [comment]
@laracasts/@jeffreyway could you have a look at #103, I think it'll be a good addition :)

**Comment by alterigor** [comment]
This feature would be really helpful!! Hope it gets added.
